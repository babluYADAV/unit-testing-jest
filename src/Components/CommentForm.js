import React, { useState } from "react";

const CommentForm = () => {
  const [text, setText] = useState("");
  const [checked, setChecked] = useState(false);
  return (
    <div>
      <h2>comment form</h2>
      <input
        type="text"
        value={text}
        placeholder="write your comment here"
        onChange={(e) => setText(e.target.value)}
      />
      <input
        type="checkbox"
        id="checkbox"
        defaultChecked={checked}
        onChange={() => setChecked(!checked)}
      />
      <label htmlFor="checkbox">i agree to terms and conditions</label>
      <button
        disabled={!checked || !text}
        onClick={() => console.log("clicked")}
      >
        Comment
      </button>
    </div>
  );
};

export default CommentForm;
