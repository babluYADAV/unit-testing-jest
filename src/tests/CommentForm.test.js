import { render, screen, fireEvent } from "@testing-library/react";
import CommentForm from "../Components/CommentForm";

describe("Commentform test", () => {
  test("initail Conditions", () => {
    render(<CommentForm />);

    const commentInput = screen.getByRole("textbox");
    expect(commentInput).toBeInTheDocument();
    const checkbox = screen.getByLabelText("i agree to terms and conditions", {
      exact: false,
    });
    expect(checkbox).toBeInTheDocument();
    const commentButton = screen.getByRole("button", {
      name: "Comment",
      exact: false,
    });
    expect(commentButton).toBeDisabled();
  });

  test("Enable submit button on type and checkbox click ", () => {
    render(<CommentForm />);
    const checkbox = screen.getByLabelText("i agree to terms and conditions", {
      exact: false,
    });
    const commentButton = screen.getByRole("button", {
      name: "Comment",
      exact: false,
    });
    const commentInput = screen.getByPlaceholderText(
      "write your comment here",
      {
        exact: false,
      }
    );
    fireEvent.change(commentInput, { target: { value: "something" } });

    fireEvent.click(checkbox);
    expect(commentButton).toBeEnabled();
    fireEvent.click(commentButton);

    fireEvent.click(checkbox);
    expect(commentButton).toBeDisabled();
  });
});
