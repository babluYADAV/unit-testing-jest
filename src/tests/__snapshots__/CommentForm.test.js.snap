// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commentform test snapshot  1`] = `
<div>
  <h2>
    comment form
  </h2>
  <input
    onChange={[Function]}
    placeholder="write your comment here"
    type="text"
    value=""
  />
  <input
    defaultChecked={false}
    id="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="checkbox"
  >
    i agree to terms and conditions
  </label>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Comment
  </button>
</div>
`;
